src/stim/py/compiled_measurement_sampler_pybind_test.py

```diff
@@ -118,6 +119,37 @@ def test_skip_reference_sample():
         [[True]],
     )
 
+def test_reference_sample_init():
+    np.testing.assert_array_equal(
+        stim.Circuit("X 0\nM 0").compile_sampler(reference_sample=None).sample(1),
+        [[True]],
+    )
+    circuit = stim.Circuit("X 0\nM 0")
+    s = stim.TableauSimulator()
+    ref_sample = np.array([False])
+    np.testing.assert_array_equal(
+        circuit.compile_sampler(reference_sample=ref_sample).sample(1),
+        [[False]],
+    )
+    ref_sample = s.reference_sample(circuit)
+    np.testing.assert_array_equal(
```

Comment:
Test that specifying a bit packed reference sample works. Make sure you have at least 9 measurements so it crosses byte boundaries. It should be the case that the 1s bit is the first measurement, the 2s bit the second, the 4s bit the fourth, etc.